core/src/avm2/verify.rs

```diff
@@ -561,279 +670,607 @@ fn optimize<'gc>(
+                // Arguments
+                for _ in 0..*num_args {
+                    stack.pop();
+                }
+
+                // Then receiver.
+                stack.pop();
+            }
+            Op::ConstructProp {
+                index: name_index,
+                num_args,
+            } => {
+                // Arguments
+                for _ in 0..*num_args {
+                    stack.pop();
+                }
+
+                // Then receiver.
+                stack.pop();
+
+                let multiname = method
+                    .translation_unit()
+                    .pool_maybe_uninitialized_multiname(*name_index, &mut activation.context);
+                if let Ok(multiname) = multiname {
+                    if multiname.has_lazy_component() {
+                        // Avoid handling lazy for now
+                        stack.clear();
+                    }
+                }
+
+                // Avoid checking return value for now
+                stack.push_any();
+            }
+            Op::CallProperty {
+                index: name_index,
+                num_args,
+            } => {
+                // Arguments
+                for _ in 0..*num_args {
+                    stack.pop();
+                }
 
-        previous_op = Some(op.clone());
+                // Then receiver.
+                stack.pop();
+
+                let multiname = method
+                    .translation_unit()
+                    .pool_maybe_uninitialized_multiname(*name_index, &mut activation.context);
+                if let Ok(multiname) = multiname {
+                    if multiname.has_lazy_component() {
+                        // Avoid handling lazy for now
+                        stack.clear();
+                    }
+                }
+
+                // Avoid checking return value for now
+                stack.push_any();
+            }
+            Op::CallPropVoid { .. } => {
+                // Avoid handling for now
+                stack.clear();
+            }
+            Op::Nop => {}
+            Op::DebugFile { .. } => {}
+            Op::DebugLine { .. } => {}
+            Op::Debug { .. } => {}
+            Op::IfTrue { .. } | Op::IfFalse { .. } => {
+                stack.pop();
+            }
+            Op::IfStrictEq { .. }
+            | Op::IfStrictNe { .. }
+            | Op::IfEq { .. }
+            | Op::IfNe { .. }
+            | Op::IfGe { .. }
+            | Op::IfGt { .. }
+            | Op::IfLe { .. }
+            | Op::IfLt { .. }
+            | Op::IfNge { .. }
+            | Op::IfNgt { .. }
+            | Op::IfNle { .. }
+            | Op::IfNlt { .. } => {
+                stack.pop();
+                stack.pop();
+            }
+            Op::Si8 => {
+                stack.pop();
+                stack.pop();
+            }
+            Op::Li8 => {
+                stack.pop();
+                stack.push_int();
+            }
+            _ => {
```

Comment:
as mentioned on discord, I think we should just have all of these checked in one go - otherwise this can't be used for verification, and can randomly inhibit optimizations.